components/post/sidebar/index.tsx

```
@@ -32,22 +35,56 @@ export default function PostSideBar({
 }: PostSideBarProps) {
-    const hasContents = contents.length > 0
-    const hasRelated = relatedPosts.length > 0
-    const hasExtras = author || Children.count(children) > 0
 
-    if (!hasContents && !hasRelated && !hasExtras) return null
+    const hasContents = contents.length > 0;
+    const hasRelated = relatedPosts.length > 0;
+    const hasExtras = author || Children.count(children) > 0;
 
+    if (!hasContents && !hasRelated && !hasExtras) return null;
+    const { handleSideBarOpenState, sidebarIsOpen, sidebarRef, openButtonRef } = useSidebar()
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)